### PR TITLE
[Hub Generated] Review request for Microsoft.Devices to add version stable/2020-01-01

### PR DESCRIPTION
**Pull request by sudeepmsft**
This is a PR generated at [OpenAPI Hub](https://aka.ms/openapihub). You can view your [work branch via this link](https://portal.azure-devex-tools.com/branch/sudeepmsft/azure-rest-api-specs/dev-deviceprovisioningservices-Microsoft.Devices-2020-01-01...master/).

### Changelog
Please ensure to add changelog with this PR by answering the following questions.
  1. What's the purpose of the update?    
      - [ ] new service onboarding 
      - [ ] new API version 
      - [ ] update existing version for new feature 
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify 
  2. When you are targeting to deploy new service/feature to public regions? Please provide date, or month to public if date is not available yet.
  3. When you expect to publish swagger? Please provide date, or month to public if date is not available yet.
  4. If it's an update to existing version,  please select SDKs of specific language and CLIs that require refresh after swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [ ] No, no need to refresh for updates in this PR

### Contribution checklist:
- [ ] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of “no breaking changes
- [ ] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [ ] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “**WaitForARMFeedback**” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  - Adding a new service

- [ ] Please ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from API Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable version with Breaking Change Validation errors
- [ ] Updating API(s) in preview over 1 year

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://msazure.visualstudio.com/One/_workitems/create/Scenario?templateId=d1c215ec-5d6a-4cb3-a193-95ddd0a1ba96&ownerId=85bb92fb-21fe-41ca-9309-47f2c5f57515). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by sudeepmsft** [comment]
@markcowl This commit is to bring this version of api doc in parity with the existing api version. Current api properties has not changed but missed these documentation during the release of that version. Current swagger docs and spec does not correctly reflect current version. So, i am just changing specs to match what API has at the moment. I would be glad to connect and clarify if necessary. Can you please review and approve.??

File: specification/deviceprovisioningservices/resource-manager/Microsoft.Devices/stable/2020-01-01/iotdps.json [hunk 1]
```diff
@@ -1418,6 +1418,11 @@
           "type": "boolean",
           "readOnly": true
         },
+        "certificate": {
+          "description": "base-64 representation of X509 certificate .cer file or just .pem file content.",
```
Comment (markcowl) [inline specification/deviceprovisioningservices/resource-manager/Microsoft.Devices/stable/2020-01-01/iotdps.json:1422]: If this is a base64-encoded string, then this should also have ```"format": "byte"```

**Review by markcowl — CHANGES_REQUESTED** [review]
Also, please fix the swagger issues other than the breaking change issue.

**Comment by sudeepmsft** [comment]
@markcowl checked in the review fixes. Added some bug links that supports these changes

Also, some of the issues that these changes fix are below..mostly **object_additional_properties**
https://portal.azure-devex-tools.com/amekpis/correctness/detail?errorId=E15D5ECF-ED67-4AF0-AA7B-26ED441DB3F9
https://portal.azure-devex-tools.com/amekpis/correctness/detail?errorId=EDA7F47F-6A55-4D53-B85D-556A6BB680DD
https://portal.azure-devex-tools.com/amekpis/correctness/detail?errorId=C067E477-8EF3-4BFC-AF6A-6FBC5B1039BE

s360 link for ref
https://vnext.s360.msftcloudes.com/blades/unifiedplatform?blade=KPI:cf751b97-2c95-420f-baa8-9a16e8643a5b~SLA:3~AssignedTo:All~Forums:All~_loc:UnifiedPlatform&def=&peopleBasedNodes=nakarim_team;nberdy_team;jasonz_team;rajeevm_team&tile=&global=4:3b7ca3aa-2894-4388-9b01-27ee2f741193;403a1a7e-9f26-44b7-be51-4c15b0d02c21

File: specification/deviceprovisioningservices/resource-manager/Microsoft.Devices/stable/2020-01-01/iotdps.json (hunk 1 above)
Comment (sudeepmsft) [reply specification/deviceprovisioningservices/resource-manager/Microsoft.Devices/stable/2020-01-01/iotdps.json:1422]: I have included the fix in the latest commit.

**Comment by sudeepmsft** [comment]
@markcowl  additional links for why the fix that caused linter issue has to be done
https://portal.azure-devex-tools.com/amekpis/correctness/detail?errorId=EDA7F47F-6A55-4D53-B85D-556A6BB680DD
https://portal.azure-devex-tools.com/amekpis/correctness/detail?errorId=587F64BE-2797-489B-8D56-80C3AEBDD9F9

**Comment by sudeepmsft** [comment]
@markcowl  fixed the lintdiff and moving it to new version. can this be merged now?

File: package-lock.json [hunk 2]
```diff
@@ -5017,7 +5017,7 @@
     "prettier": {
```
Comment (markcowl) [inline package-lock.json:5017]: this should not be included in the pr

**Review by markcowl — CHANGES_REQUESTED** [review]
Remove this one file from your change, otherwise, lgtm

File: package-lock.json [hunk 3]
```diff
@@ -1,6724 +0,0 @@
-{
```
Comment (markcowl) [inline package-lock.json:1]: You should not delete this file, you should just remove any changes to this file from your commit.  You can checkout the version fo the file from the master branch if necessary.

**Review by markcowl — CHANGES_REQUESTED** [review]

**Review by markcowl — APPROVED** [review]
